test/common/upstream/load_balancer_fuzz.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package test.common.upstream;
+
+import "validate/validate.proto";
+import "envoy/config/cluster/v3/cluster.proto";
+import "google/protobuf/empty.proto";
+
+message UpdateHealthFlags {
+  uint64 host_index = 1;
+  uint32 num_healthy_hosts = 2;
+  uint32 num_degraded_hosts = 3;
+  uint32 num_excluded_hosts = 4;
+}
+
+message LbAction {
+  oneof action_selector {
+    option (validate.required) = true;
+    UpdateHealthFlags update_health_flags = 1;
+    google.protobuf.Empty prefetch = 2;
+    google.protobuf.Empty choose_host = 3;
+  }
+}
+
+//This message represents what LoadBalancerFuzzBase will interact with, performing setup of host sets and calling into load balancers.
+//The logic that this message represents and the base class for load balancing fuzzing will be logic that maps to all types of load balancing
+//and can be modularly used at the highest level for each load balancer.
+message LoadBalancerTestCase {
+  envoy.config.cluster.v3.Cluster.CommonLbConfig common_lb_config = 1
+      [(validate.rules).message.required = true];
+  repeated LbAction actions = 2;
+
+  //TODO: Localities
+  repeated uint32 num_hosts_in_host_set = 3
+      [(validate.rules).repeated = {min_items: 1, max_items: 20}];
+
+  //This number is used to instantiate the prng. The prng takes the place of random() calls, allowing a representative random distribution which is also deterministic.
```

Comment:
I'm actually curious on how fuzzer efficacy changes when you use a PRNG seed vs. explicit collection of entropy under the control of the fuzzer. Can you run any local experiments to compare this and what you had before? This would be something useful to learn about, because if explicit entropy is better, then we should make that a general fuzz library feature. If not, the conciseness of a PRNG seed seems a win.

Reply:
Thinking about this a bit more. I think what you have in this problem is a need to consume both structured (protobuf) and unstructure (a series of bytes to drive various selection criteria). Do you really need the latter, or can everything be represented in a structured way?

Reply:
I think what was so nice about the prng is the fact that it kind of "scaled" with the amount of random calls needed. The fuzzing library would generate potential very long action streams (although a hard cap at 64), so this prevents the fuzzer creating like 50 64 bit chunks of randomness on a fuzz iteration that only needed one random call.

Reply:
I think that's the biggest thought I had, also say the fuzzing engine generates 2 64 bit chunks of randomness and you have something like 50 random() calls. This would simply go back and forth between the two 64 bit chunks, not representing a truly random distribution. What's great about the prng way you suggestion was the it "scales" alongside the random() calls, whether a lot are needed or a few are needed.

Reply:
> Do you really need the latter, or can everything be represented in a structured way?

This would be feel like the prior solution of the repeated uint64_ts that would correspond to randomgenerator's return values, which (thinking about the subset selection algorithm) would make it very easy for the fuzzer to module return values. OTOH, we might trust/believe RandomGenerator to truly be sampling from a random distribution. If so, what's the point of "fuzzing" the randomness? A PRNG would model that distribution fairly well.

Reply:
+1 to Asra's comment, agree completly.